pandas/core/indexes/base.py

```diff
@@ -3221,7 +3222,15 @@ def _convert_arr_indexer(self, keyarr):
         -------
         converted_keyarr : array-like
         """
-        keyarr = com.asarray_tuplesafe(keyarr)
+
+        if (
+            isinstance(keyarr, list)
```

Comment:
what are you trying to do?

Reply:
@jreback 
I try to infer list of strings are representing datetimes. I changed code but changed code also have same reason.